Documentation/design-docs/arm64-intrinsics.md

```diff
@@ -0,0 +1,217 @@
+## Arm64 Intrinsics
+
+This document is intended to document proposed design decision related to the introduction
+of Arm64 Intrinsics
+
+### Intrinsics in general
+
+Use of intrinsics in general is a CoreCLR design decision to allow low level platform
+specific optimizations.  
+
+At first glance, such a decision seems to violate the fundamental principles of .NET
+code running on any platform.  However, the intent is not for the vast majority of
+apps to use such optimizations.  The intended usage model is to allow library
+developers access to low level functions which enable optimization of key
+functions.  As such the use is expected to be limited, but performance critical.
+
+#### Intrinsic `IsSupported` granularity
+
+The existing implementation is using `IsSupported` to determine whether an existing
+namespace is supported on a given platform.
+
+The granularity is kept at a large scale to have large blocks of functionality controlled
```

Comment:
There seems to be a need to start explaining differences between Intel and Arm intrinsics implementation at this stage. Currently we have IsSupported property mapping defined at CPUID feature granularity for Intel intrinsics. The question is  how this mapping will look like for Arm intrinsics given significantly wider range of CPUs implementing ISA.

Another area which needs consideration is a new intrinsics extensions specification support. It was released as a EAC (out of beta) on August 31, 2017 under the title [The Scalable Vector Extension (SVE), for ARMv8-A](https://developer.arm.com/docs/ddi0584/latest/arm-architecture-reference-manual-supplement-the-scalable-vector-extension-sve-for-armv8-a). We should expect support for vectors ranging from 128bits to 2049bits in 128bits increments. Do we need IsSupported to detect what vector sizes will be supported on Vector256/Vector384/Vector512/Vector640/Vector768/ etc sturcts?

Reply:
The actual `Vector*` types have their own `IsSupported` flag today, so maybe that can be reused in combo with one of Arm intrinsic classes (not familiar with how these are exposed myself)?

Reply:
`Vector<T>` has an `IsHwAccelerated` flag.  

X86 is implementing each IsSupported as a separate hw intrinsic.  I will propose ARM64 uses a slightly different approach